torchbox/torchbox.com
# Navigation - BE

[Link to Ticket](https://torchbox.monday.com/boards/1192293412/pulses/1383594351)

### Description of Changes Made

This updates the primary navigation setup so it works according to the [new design](https://www.figma.com/file/QdMXBHr57mN1svsaNQ7Ggj/Torchbox-2024-(Master)?type=design&node-id=2850-40823&mode=design&t=6FQmSsdAb6gyA4DO-4), which allows for a multi-level menu (up to 3 levels).

The initial approach was to modify the primary nav settings to allow for a nested structure. However, it was observed that this could become complicated and may in turn be confusing for content editors (please see ticket overview for link to Slack thread). Therefore, the existing flat structure of the primary nav settings is maintained, and changes are instead made to the templates to control the rendering of the primary navigation.

### How to Test

- Add some links to the primary navigation in the *Navigation Settings*. These should be displayed on the frontend (without the final styling)
- For any of the pages added above with children and grandchildren, edit a few of the children/granchildren by checking the _Show in Menu_ box in the promote tab. These should appear on the frontend. (please see screenshot below for reference)

### Screenshots

<details>
  <summary>Expand to see more</summary>

![image](https://github.com/torchbox/torchbox.com/assets/7713776/098788d9-b072-4483-81f4-d1d97a0cca1f)

</details>

### MR Checklist


#### Unit tests


#### Documentation


#### Browser testing

  - Latest version of Chrome on Linux
  - Latest version of Firefox on Linux

#### Data protection


#### Accessibility


#### Sustainability


#### Pattern library

- [ ] ~Changes are not relevant the pattern library~ The pattern library will be updated on the FE ticket, as the templates will change significantly from the current version.
